cmd/trace-agent/config/config.go

```diff
@@ -597,3 +594,55 @@ func acquireHostnameFallback(c *config.AgentConfig) error {
 	log.Debugf("Acquired hostname from core agent (%s): %q.", c.DDAgentBin, c.Hostname)
 	return nil
 }
+
+//UpdateConfigHandler returns handler for configuration changes during runtime
+func UpdateConfigHandler() http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		_ = req.ParseForm()
+
+		setting := html.UnescapeString(req.Form.Get("setting"))
+		value := html.UnescapeString(req.Form.Get("value"))
+
+		log.Warnf("Got a request to change a setting: %s", setting)
+
+		switch setting {
+		case "log_level":
+			currentLvl, err := log.GetLogLevel()
+			if err != nil {
+				log.Errorf("Couldn't get log level: %s", err.Error())
+				httpSettingError(w, http.StatusBadRequest, err)
+				return
+			}
+
+			seelogLogLevel := strings.ToLower(value)
+			if seelogLogLevel == "warning" {
+				seelogLogLevel = "warn"
+			}
+
+			if seelogLogLevel == currentLvl.String() {
+				httpSettingError(w, http.StatusBadRequest,
+					errors.New("current log level equals requested value"))
+				return
+			}
+
+			if err = coreconfig.ChangeLogLevel(seelogLogLevel); err != nil {
+				httpSettingError(w, http.StatusBadRequest, err)
+				return
+			}
+			coreconfig.Datadog.Set("log_level", seelogLogLevel)
+
+			log.Infof("Switched log level from %s to %s", currentLvl, seelogLogLevel)
+		default:
+			httpSettingError(w, http.StatusBadRequest, errors.New("unrecognized setting"))
+			return
+		}
+
+		w.Write([]byte("Success"))
+
+		return
+	})
+}
+func httpSettingError(w http.ResponseWriter, status int, err error) {
```

Comment:
Can we move this to `pkg/trace/api/responses.go` and call it `httpError` ?

Reply:
As an exported HTTPError, sure

Reply:
Oh, sorry. My bad. I got confused with the packages having moved around. You can keep it here, perhaps changing to the name to `httpError`.

Reply:
Okay